tests/dataset/test_aoi.py

```diff
@@ -63,43 +71,56 @@ def test_singleband_input(self):
         """Tests reading a singleband raster as input with ${dataset.bands} pattern"""
         extract_archive(src="tests/data/spacenet.zip")
         data = read_csv("tests/tiling/tiling_segmentation_binary-singleband_ci.csv")
-        for row in data:
-            aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'], raster_bands_request=['R', 'G', 'B'])
-            aoi.close_raster()
+        bands = ['R', 'G', 'B']
+        row = next(iter(data))
+        aoi = AOI(raster=row['tif'], label=row['gpkg'], split=row['split'], raster_bands_request=bands)
+        assert str(aoi.raster_name) == aoi.raster_raw_input.replace("${dataset.bands}", "band")
+        assert aoi.raster_name.stem == aoi.aoi_id
+        assert aoi.raster_bands_request == bands
+        aoi.close_raster()
 
     def test_stac_input(self):
         """Tests singleband raster referenced by stac item as input"""
         extract_archive(src="tests/data/spacenet.zip")
         data = read_csv("tests/tiling/tiling_segmentation_binary-stac_ci.csv")
-        for row in data:
-            aoi = AOI(
-                raster=row['tif'], label=row['gpkg'], split=row['split'],
-                raster_bands_request=['red', 'green', 'blue'])
-            aoi.close_raster()
+        bands = ['red', 'green', 'blue']
+        row = next(iter(data))
+        aoi = AOI(
+            raster=row['tif'], label=row['gpkg'], split=row['split'],
+            raster_bands_request=bands)
+        assert str(aoi.raster_name) == aoi.raster_raw_input
+        assert aoi.raster_name.stem == aoi.aoi_id
+        assert aoi.raster_bands_request == bands
+        for band, actual_raster in zip(["R", "G", "B"], aoi.raster_parsed):
+            root_raster = "http://datacube-stage-data-public.s3.ca-central-1.amazonaws.com/store/imagery/optical/" \
+                          "spacenet-samples/"
+            exp_raster = root_raster + f'SpaceNet_AOI_2_Las_Vegas-056155973080_01_P001-WV03-{band}.tif'
```

Comment:
Improved this test by checking expected vs actual raster files as found through the initial stac item